sematic/retry_settings.py

```diff
@@ -28,10 +28,13 @@ def __post_init__(self):
         if not isinstance(self.exceptions, (tuple, list, set)):
             raise ValueError(f"exceptions should be a tuple, got {self.exceptions}")
 
-    def should_retry(self, exception_metadata: ExceptionMetadata) -> bool:
+    def should_retry(self, exception_metadata: Optional[ExceptionMetadata]) -> bool:
         """
         Should the given exception trigger a retry?
         """
+        if exception_metadata is None:
+            return True
```

Comment:
This was actually a hack I was unhappy with, caused by the fact that the 2 different exception metadata were checked one-at-a-time.
Instead of that, I rewrote `should_retry` to check an arbitrary number of exception metadata at once, and return True if any of them matches.